content/Snakemake_intro.md

```diff
@@ -12,27 +26,471 @@ Example usecase from GUI -> multiple scripts run manually -> bash script to run
+            "results/word_count.txt",
+            category="Step 2"
+        )
+    shell:
+        "wc -w {input} > {output}"
+
+```
+Here, we included both text files in the report. After opening it, we can download them.
+
+
+## Monitoring
+
+Warning: this section may not work for Windows users!
+
+Snakemake offers monitoring feature. In this case, monitoring means have live insights into a workflow execution progress. Monitoring is offer via external tool called `panoptes`. It is a simple web page with an API to which Snakemake connects. It has to be deployed, but the process is easy.
+```bash
+pip install panoptes-ui
+```
+
+To run the server on the localhost (i.e. on our computer), we can just run the following command:
+```bash
+panoptes
+```
+To close the server, we can use `CTRL+C` in the terminal. The server is running by default on `http://127.0.0.1:5000`. If we open our browser and visit this website, we will see the graphical interface for monitoring workflows. Note that now we cannot use our terminal, because it is used by `panoptes`. We have to open a new one to be able to run new commands. To connect Snakemake to the monitoring, we modify our `snakemake` command:
+```bash
+snakemake --cores 1 --wms-monitor http://127.0.0.1:5000
+```
+If you server has a different URL, replace it in the command above. When we execute the workflow, we can see some basic statistics about the execution.
+
+Warning: this set-up might be not secure enough for the production use. See [this thread](https://github.com/panoptes-organization/panoptes/issues/172) for more details.
+
+
+## Workflow reproducibility
+
+Reproducibility in computational workflows ensures that an analysis can be consistently repeated with the same results, which is crucial for validating findings, ensuring transparency, and facilitating collaboration. It allows other researchers to verify results, understand methodologies, and build upon previous work.
+
+### Project structure 
+
+Snakemake provides recommendations regarding project structure. 
+```
+├── .gitignore
+├── README.md
+├── LICENSE.md
+├── workflow
+│   ├── rules
+|   │   ├── module1.smk
+|   │   └── module2.smk
+│   ├── envs
+|   │   ├── tool1.yaml
+|   │   └── tool2.yaml
+│   ├── scripts
+|   │   ├── script1.py
+|   │   └── script2.R
+│   ├── notebooks
+|   │   ├── notebook1.py.ipynb
+|   │   └── notebook2.r.ipynb
+│   ├── report
+|   │   ├── plot1.rst
+|   │   └── plot2.rst
+|   └── Snakefile
+├── config
+│   ├── config.yaml
+│   └── some-sheet.tsv
+├── results
+└── resources
+```
+
+Following the [documentation](https://snakemake.readthedocs.io/en/stable/snakefiles/deployment.html#distribution-and-reproducibility): 
+> The workflow code goes into a subfolder workflow, while the configuration is stored in a subfolder config. Inside of the workflow subfolder, the central Snakefile marks the entrypoint of the workflow (it will be automatically discovered when running snakemake from the root of above structure).
```

Comment:
```suggestion
> The workflow code goes into a subdirectory called `workflow`, while the configuration is stored in a subdirectory called `config`. Inside of the `workflow` subdirectory, the central `Snakefile` marks the entrypoint of the workflow (it will be automatically discovered when running snakemake from the root of above structure).
```